### PR TITLE
Remove deprecated tomcat configuration file instead of moving it, sin…

### LINKED ISSUE #2537: Fix Jetty configuration file bugs during Upgrade from 4.x to 4.11
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
packaging
~~~

##### CLOUDSTACK VERSION

~~~
4.11
master
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

The `packaging/centos7/cloud.spec` file line 400 moves the `%{_sysconfdir}/sysconfig/%{name}-management` file to `%{_sysconfdir}/default/%{name}-management`.
This is unnecessary because  the file contains the deprecated tomcat configurations. Moving it prevents the 4.11 upgrade from overwriting the old file with the new jetty configurations file. 

Also `systemctl daemon-reload` command needs to be added to the end of the `%post` management step in order to refresh daemon services so that management server can start up seamlessly after the upgrade.

##### STEPS TO REPRODUCE

To Reproduce: install version 4.6 of cloudstack. 
Update Systemvm templates. http://docs.cloudstack.apache.org/projects/cloudstack-release-notes/en/4.11.0.0/upgrade/upgrade-4.5.html#update-system-vm-templates
Stop management server and usage server.
Do database backup.
Change the /etc/yum.repos.d/cloudstack.repo file to upgrade to 4.11.
Do sudo yum upgrade cloudstack-management.
After upgrade it fails to start the management server because:
1) the less /etc/default/cloudstack-management file still contains the old tomcat configurations and not the new 4.11 Jetty server configurations.
2) the systemctl daemon-reload needs to be executed to relaod the daemon services

~~~
systemctl stop cloudstack-management
systemctl stop cloudstack-usage
mysqldump -u root -p cloud > cloud-backup_`date '+%Y-%m-%d'`.sql
mysqldump -u root -p cloud_usage > cloud_usage-backup_`date '+%Y-%m-%d'`.sql
vi /etc/yum.repos.d/cloudstack.repo
sudo yum upgrade cloudstack-management
sudo yum upgrade cloudstack-usage
systemctl start cloudstack-management
systemctl start cloudstack-usage
~~~

##### EXPECTED RESULTS

~~~
after upgrade:
less /etc/default/cloudstack-management should show the new jetty server environment variables
management server should start up successfully without manually executing systemctl daemon-reload
able to log in to the UI  
~~~

##### ACTUAL RESULTS

management server failed to start and returned an error message "Job for cloudstack-management.service failed because the control process exited with error code. ..."

~~~
systemctl start cloudstack-management produced the above error output
~~~

### PR DESCRIPTION
**Pull request by ernjvr**
…ce moving it prevented the 4.11 upgrade from overwriting it with the new jetty configurations. Also added systemctl daemon-reload to refresh daemon services so that management server can start up seamlessly after the upgrade.

## Description

This fixes #2537

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [ ] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

@blueorangutan package

### CODE REVIEW

**Comment by ernjvr** [comment]
Manually tested the upgrade from 4.6 to 4.11 and verified:
- /etc/sysconfig/cloudstack-management file has been removed
- /etc/default/management-server has the correct version 4.11 configurations
- management server starts successfully
- UI login is successful
- all VMs started successfully

**Comment by rhtyd** [comment]
@blueorangutan package

File: packaging/centos63/cloud.spec [hunk 1]
```diff
@@ -454,8 +454,14 @@ if [ -f "%{_sysconfdir}/cloud.rpmsave/management/db.properties" ]; then
     mv %{_sysconfdir}/cloud.rpmsave/management/db.properties %{_sysconfdir}/cloud.rpmsave/management/db.properties.rpmsave
 fi
 
+if [ -f %{_sysconfdir}/sysconfig/%{name}-management ] ; then
+    rm -f %{_sysconfdir}/sysconfig/%{name}-management
+fi
+
 chown -R cloud:cloud /var/log/cloudstack/management
 
+systemctl daemon-reload
```
Comment (rhtyd) [inline packaging/centos63/cloud.spec]: @ernjvr remove the systemctl thing, CentOS6/el6 does not have systemd. No need to reload anything here. Rest is LGTM.

**Comment by ernjvr** [comment]
@blueorangutan package

**Comment by ernjvr** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
LGTM, travis okay. I'll merge this.

**Comment by rhtyd** [comment]
The failed tests are all env related, the changes in the PR cannot be validated with Trillian smoke tests as they are upgrade related. Given the upgrade procedure was manually tested by @ernjvr / @borisstoyanov I'm okay with the merging, no regressions seen. We'll re-test during weekly runs.

**Pull request merged**